pkg/kubelet/cm/cpumanager/state/state_mem.go

```diff
@@ -23,20 +23,39 @@ import (
 	"k8s.io/kubernetes/pkg/kubelet/cm/cpuset"
 )
 
+type ContainerCpuAssignment map[string]cpuset.CPUSet
+
+func (as ContainerCpuAssignment) DeepCopy() ContainerCpuAssignment {
+	ret := make(ContainerCpuAssignment)
+	for key, val := range as {
+		ret[key] = val
+	}
+	return ret
+}
+
+
 type stateMemory struct {
 	sync.RWMutex
-	assignments   map[string]cpuset.CPUSet
+	assignments   ContainerCpuAssignment
 	defaultCPUSet cpuset.CPUSet
+	stateFile     *stateFile
 }
 
 var _ State = &stateMemory{}
 
 // NewMemoryState creates new State for keeping track of cpu/pod assignment
 func NewMemoryState() State {
-	glog.Infof("[cpumanager] initializing new in-memory state store")
+	glog.Infof("[cpumanager] state memory: initializing new state store")
```

Comment:
Let's leave the memory-backed state implementation as-is and just implement a new file-backed state.

Reply:
+1